### ISSUE TITLE
Web3-Mock doesn't work with the `@metamask/test-dapp`

### ISSUE THREAD

**Issue by matstyler** [DePayFi/web3-mock#34]
I was able to setup `Web3-Mock` using Playwright but I cannot use it properly with Cypress. Lets say I have a very simple use case. I mock the metamask wallet and then I want to test that mock using `@metamask/test-dapp`. Nothing is working as expected and I get provider errors:

![Screenshot 2024-06-02 at 15 59 22](https://github.com/DePayFi/web3-mock/assets/63294765/f939380f-3c96-43b2-b6ac-afe26db7f56d)

The thing is I am able to use `window` object in the e2e test code but this is not what I need. I want the mock working in e2e manual test - for example by clicking on the "ETH ACCOUNTS" button.

![Screenshot 2024-06-02 at 16 01 00](https://github.com/DePayFi/web3-mock/assets/63294765/9406dbc2-4339-4147-bf30-336de079e3d5)

**Comment by johanneskares** [comment]
Yeah same here, I created a new lib to be able to use a mock wallet, not Metamask: https://github.com/johanneskares/wallet-mock

**Comment by 0xNe0x1** [comment]
If you call `visit` after `mock` the entire javascript state gets flushed. Meaning the entire mock will be deleted.

Try
```
cy.visit("http://localhost:9999");
mock();
cy.get("#getAccounts").click();
```
